### DISCUSSION TITLE
The number of servers in the master service must always be odd

### DISCUSSION THREAD

**Discussion by cchai-newtrek** [seaweedfs/seaweedfs#5744] [Q&A] [answered]
The number of servers in the master service must always be odd, I cannot understand why it should be? So if I start 3 master service, then one is go down, then I'll left 2 master services then (which is even), and it still can elect the leader in 2 master service, so what is the point of having odd number of master services then?

**Comment by vrvico** [comment, marked as answer]
Its Raft's quorum: (N/2)+1.
2 masters: Quorum is 2. If leader fails, quorum can never be achieved to elect the second master as a leader. Service is now down. So having two masters is useless.
3 masters: Quorum is 2. If leader fails, quorum elects another leader. Tolerate loss of 1 master and service is still up.
4 masters: Quorum is 3. If leader fails, quorum elects another leader. Tolerate loss of 1 master and service is still up. If second master fails, quorum will never be achieved. So having 4 masters instead of just 3 is useless.
5 masters: Quorum is 3. If leader fails, quorum elects another leader. Tolerate loss of 2 masters and service is still up....

**Reply by cchai-newtrek** [reply]
Thx for your detail explanation, so do you think this slightly modification for the Raft's quorum work?

For me, it is quite strange that as you know the leader is out, why bother still counting it in the quorum?

Instead of using (N/2) +1 quorum, use (N'/2)+1 quorum where N' = N-1, which is the remaining number of master after the leader go down?

Or is it difficult to implement?

> Its Raft's quorum: (N/2)+1. 2 masters: Quorum is 2. If leader fails, quorum can never be achieved to elect the second master as a leader. Service is now down. So having two masters is useless. 3 masters: Quorum is 2. If leader fails, quorum elects another leader. Tolerate loss of 1 master and service is still up. 4 masters: Quorum is 3. If leader fails, quorum elects another leader. Tolerate loss of 1 master and service is still up. If second master fails, quorum will never be achieved. So having 4 masters instead of just 3 is useless. 5 masters: Quorum is 3. If leader fails, quorum elects another leader. Tolerate loss of 2 masters and service is still up....
